Documents/AnalyticsToolDesignDocument.md

```diff
@@ -0,0 +1,207 @@
+- **Modified paths** - The number of files which have been edited and have been handed over to git. This list may contain just directories but not a reference to all of their children who will accordingly also all have modified paths. The command `git ls-tree -r HEAD` can be used to access the file system without risking unintentionally hydrating files
+- **Git index** - The differences between modified paths and the index tracked by git. This might be useful in determining what differences the local repository has to predict the complexity and potential slowdowns of an upcoming commit.
+- **Placeholders** - A total number of the placeholders being tracked in VFS4G and a list of the placeholders in the system. The list VFS4G maintains should always track every placeholder object on disk
+- **Troublesome programs** - Similar to what is already being logged in the heartbeat, look at which services or programs are responsible for hydrating large amounts of files in the enlistment
+- **Command run-times** - A history of the time it takes for git commands to run. This can include averages, trends, and eventually graphs (implemented by the client in the visualization component)
+
+## Components
+
+- **API** - The server component of the client-server relationship. Uses the named pipe to hand rudimentary data to the verb for processing and computation. It sits within the `InProgressMount` process and interacts with the VFS4G internals
+- **Verb** - The `gvfs statistics` verb will be the component that handles the client side interaction with the API. When run, it will collect and store the statistics from the enlistment to either be displayed immediately or collected for trend purposes to then provide understanding in the future. This is also the point where data can be sent to telemetry for our internal analysis. The verb will also output the data to the console with a tool to help visualize and understand the data. It will output the analytics data to the command line and will also be used during the design period to test the API and verb
+- **Utility tools for dehydration** - *Stretch goal* | After the analytics are collected, these components will provide utilities to help the user get their enlistment back to a healthier state. The ideal goal is to have VFS4G suggest uses to the user (EG: `This directory was hydrated by Sublime text editor. You can dehydrate it by running "gvfs dehydrate --confirm -d /path/to/repo"`)
+- **Additional visualization tools** - *Stretch goal* | With a comprehensive API in place, the possibility opens up for different visualization tools to be created that consume the json data and display it to the user in different ways, potentially even eventually interfacing with the utility tools for an interactive GUI
+- **Alert system** - *Stretch goal* | Auto run the statistics command after an enlistment was automatically mounted and suggest action if the hydration levels of the repository seem high
+
+## CLI Tool Use Cases

```

Comment:
The document update is probably going to be a second round of stuff since a lot of things have changed. Since we're towards finished with this feature, I plan to make the remaining appropriate changes and then after I'm done here I'll overhaul the document accordingly